src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs

```diff
@@ -116,69 +117,80 @@ public override void ConfigureServices(IServiceCollection services)
 
         public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
         {
-            var service = serviceProvider.GetRequiredService<IOpenIdServerService>();
-            var settings = serviceProvider.GetRequiredService<IOptions<Settings.OpenIdServerSettings>>().Value;
-            var validationResult = service.ValidateSettingsAsync(settings).GetAwaiter().GetResult();
+            var servic = serviceProvider.GetRequiredService<IOpenIdServerService>();
+            var service = serviceProvider.GetRequiredService<ISiteService>();
+            
+            var settings = service.GetSiteSettingsAsync()
+                .GetAwaiter()
+                .GetResult()
+                .As<Settings.OpenIdServerSettings>();
 
-            if (validationResult.Length == 0)
+            if (settings != null)
             {
-                var openIddictServerOptions = serviceProvider
-                    .GetRequiredService<IOptionsMonitor<OpenIddictServerOptions>>()
-                    .Get(OpenIddictServerDefaults.AuthenticationScheme);
+                var validationResult = servic.ValidateSettingsAsync(settings)
+                    .GetAwaiter()
+                    .GetResult();
 
-                if (openIddictServerOptions.AuthorizationEndpointPath != PathString.Empty)
+                if (validationResult.Length == 0)
```

Comment:
I would have thought that only this line would be sufficient to fix the issue.

Reply:
The admin view also uses the same mechanic, so I had to rewrite that.